File: pilhas-filas/javascript/queue1.js
export default class Queue1 {
    constructor() {
        this.list = [];
    }

    isEmpty() {
        return this.list.length === 0;
    }

    add(item) {
        this.list.push(item);
    }

    remove() {
        if (this.isEmpty()) {
            throw new Error("Queue is empty");
        }
        return this.list.shift();
    }

    peek() {
        if (this.isEmpty()) {
            throw new Error("Queue is empty");
        }
        return this.list[0];
    }

    count() {
        return this.list.length;
    }
}

File: pilhas-filas/csharp/queue1-test/Queue1.cs
using System;
using System.Collections.Generic;

public class Queue1<T>
{
    private List<T> list;

    public Queue1()
    {
        this.list = new List<T>();
    }

    public bool IsEmpty()
    {
        return this.list.Count == 0;
    }

    public void Add(T item)
    {
        this.list.Add(item);
    }

    public T Remove()
    {
        if (this.IsEmpty())
        {
            throw new InvalidOperationException("Queue is empty");
        }
        T item = this.list[0];
        this.list.RemoveAt(0);
        return item;
    }

    public T Peek()
    {
        if (this.IsEmpty())
        {
            throw new InvalidOperationException("Queue is empty");
        }
        return this.list[0];
    }

    public int Count
    {
        get
        {
            return this.list.Count;
        }
    }
}
